File: src/main/java/net/darkmorford/jas/item/ItemBlockSnadMeta.java
package net.darkmorford.jas.item;

import net.minecraft.block.Block;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;

public class ItemBlockSnadMeta extends ItemBlock {

	public ItemBlockSnadMeta(Block block) {
		super(block);
		if (!(block instanceof IMetaBlockSnad)) {
			throw new IllegalArgumentException(String.format("The given Block %s is not an instance of IMetaBlockSnad!",
					block.getUnlocalizedName()));
		}
		setMaxDamage(0);
		setHasSubtypes(true);
	}

	@Override
	public int getMetadata(int damage) {
		return damage;
	}

	@Override
	public String getUnlocalizedName(ItemStack stack) {
		return String.format("%s.%s", super.getUnlocalizedName(stack), ((IMetaBlockSnad) block).getSpecialName(stack));
	}
}

File: src/main/kotlin/net/darkmorford/jas/item/ItemBlockSnadMeta.kt
package net.darkmorford.jas.item

import net.minecraft.block.Block
import net.minecraft.item.ItemBlock
import net.minecraft.item.ItemStack

class ItemBlockSnadMeta(block: Block) : ItemBlock(block) {
	init {
		if (block !is IMetaBlockSnad) {
			throw IllegalArgumentException(String.format(
					"The given Block %s is not an instance of IMetaBlockSnad!",
					block.unlocalizedName))
		}
		maxDamage = 0
		hasSubtypes = true
	}

	override fun getMetadata(damage: Int): Int {
		return damage
	}

	override fun getUnlocalizedName(stack: ItemStack): String {
		return "${super.getUnlocalizedName(stack)}.${(block as IMetaBlockSnad).getSpecialName(stack)}"
	}
}


